### PR TITLE
Minor changes to enable Linux testing

### LINKED ISSUE #3658: Persisting the credentials while running EntityFramework.MicrosoftSqlServer.FunctionalTests
While running the tests in EntityFramework.MicrosoftSqlServer.FunctionalTests after changing the config.json to point to a SqlServer, the password is not used for all test runs. 
As a result, many tests throw a login failed exception.

Can this be modified to either persist the credentials or to use the credentials from config.json everytime the tests are run?

### PR DESCRIPTION
**Pull request by natemcmaster**
Handle line endings in query tests.

Fix #3658

### CODE REVIEW

**Comment by natemcmaster** [comment]
cc @saurabh500

File: test/EntityFramework.MicrosoftSqlServer.FunctionalTests/QuerySqlServerTest.cs [hunk 1]
```diff
@@ -4077,6 +4077,9 @@ public override void Selected_column_can_coalesce()
                 Sql);
         }
 
-        private static string Sql => TestSqlLoggerFactory.Sql;
+        private static string FileLineEnding = @"
+";
+
+        private static string Sql => TestSqlLoggerFactory.Sql.Replace(Environment.NewLine, FileLineEnding);
     }
```
Comment (smitpatel) [inline test/EntityFramework.MicrosoftSqlServer.FunctionalTests/QuerySqlServerTest.cs]: What is the reason for this replacement?

Comment (natemcmaster) [inline test/EntityFramework.MicrosoftSqlServer.FunctionalTests/QuerySqlServerTest.cs]: LF vs CRLF. The logger uses Environment.NewLine but git ensures this file is CRLF even on Linux.

**Comment by bricelam** [comment]
Clever. :smile:

File: test/EntityFramework.MicrosoftSqlServer.FunctionalTests/Utilities/SqlConnectionStringBuilderExtensions.cs [hunk 2]
```diff
@@ -41,6 +42,11 @@ public static SqlConnectionStringBuilder ApplyConfiguration(this SqlConnectionSt
             if (!string.IsNullOrEmpty(config["Password"]))
             {
                 builder.Password = config["Password"];
+
+                if (Type.GetType("Mono.Runtime") == null)
+                {
+                    builder.PersistSecurityInfo = true;
+                }
```
Comment (smitpatel) [inline test/EntityFramework.MicrosoftSqlServer.FunctionalTests/Utilities/SqlConnectionStringBuilderExtensions.cs]: What is different on mono?

**Comment by bricelam** [comment]
:shipit:

File: test/EntityFramework.MicrosoftSqlServer.FunctionalTests/Utilities/SqlConnectionStringBuilderExtensions.cs (hunk 2 above)
Comment (natemcmaster) [inline test/EntityFramework.MicrosoftSqlServer.FunctionalTests/Utilities/SqlConnectionStringBuilderExtensions.cs]: Setting builder.PersistSecurityInfo throws "NotImplementedException". PersistSecurityInfo is always true on Mono.

File: test/EntityFramework.MicrosoftSqlServer.FunctionalTests/QuerySqlServerTest.cs (hunk 1 above)
Comment (roji) [inline test/EntityFramework.MicrosoftSqlServer.FunctionalTests/QuerySqlServerTest.cs]: I had some newline-related issues as well (see #3116).

Shouldn't this trick be done in all tests?

Comment (natemcmaster) [inline test/EntityFramework.MicrosoftSqlServer.FunctionalTests/QuerySqlServerTest.cs]: @roji We'll need to add this trick to all files that use multiline verbatim strings. These small changes were focused only on getting the SQL Server testing working for local dev work. We will need to make many more changes to fully automate these tests to work on Linux. (See #3594 )

**Pull request merged**